calico_kubernetes/calico_kubernetes.py

```diff
@@ -604,18 +690,26 @@ def _label_to_tag(self, label_key, label_value):
     logger.info("Using ETCD_AUTHORITY=%s", os.environ[ETCD_AUTHORITY_ENV])
     logger.info("Using CALICOCTL_PATH=%s", CALICOCTL_PATH)
     logger.info("Using KUBE_API_ROOT=%s", KUBE_API_ROOT)
+    logger.info("Using CALICO_IPAM=%s", CALICO_IPAM)
     mode = sys.argv[1]
 
-    if mode == 'init':
-        logger.info('No initialization work to perform')
-    else:
-        # These args only present for setup/teardown.
-        namespace = sys.argv[2].replace('/', '_')
-        pod_name = sys.argv[3].replace('/', '_')
-        docker_id = sys.argv[4]
-        if mode == 'setup':
-            logger.info('Executing Calico pod-creation hook')
-            NetworkPlugin().create(namespace, pod_name, docker_id)
-        elif mode == 'teardown':
-            logger.info('Executing Calico pod-deletion hook')
-            NetworkPlugin().delete(namespace, pod_name, docker_id)
+    try:
+        if mode == 'init':
+            logger.info('No initialization work to perform')
+        else:
+            # These args only present for setup/teardown.
+            namespace = sys.argv[2].replace('/', '_')
+            pod_name = sys.argv[3].replace('/', '_')
+            docker_id = sys.argv[4]
+            if mode == 'setup':
+                logger.info('Executing Calico pod-creation hook')
+                NetworkPlugin().create(namespace, pod_name, docker_id)
+            elif mode == 'teardown':
+                logger.info('Executing Calico pod-deletion hook')
+                NetworkPlugin().delete(namespace, pod_name, docker_id)
+            elif mode == 'status':
+                logger.info('Executing Calico pod-status hook')
+                NetworkPlugin().status(namespace, pod_name, docker_id)
+    except:
```

Comment:
I know we talked about this - in general it is bad to catch all exceptions.  In this case, it is OK, but we should add a comment explaining why we're doing this.

This is an "exception" to that rule, since we're re-raising the exception right after - we're just capturing it so that we can get the log outpupt.